src/Math/tests/SplineTest.cpp

```diff
@@ -154,38 +189,40 @@ TEST_CASE("Cubic spline")
 
     constexpr std::chrono::nanoseconds dT = (finalTime - initTime) / (numberOfPoints - 1);
     std::array<Eigen::Vector4d, 6> coefficients;
-    for(auto& coeff: coefficients)
+    for (auto& coeff : coefficients)
     {
         coeff.setRandom();
     }
 
-    std::vector<Eigen::VectorXd> knots;
+    std::vector<Eigen::Vector4d> knots;
     std::vector<std::chrono::nanoseconds> time;
     for (std::size_t i = 0; i < numberOfPoints; i++)
     {
         time.push_back(dT * i + initTime);
         const double t = std::chrono::duration<double>(time.back()).count();
-        knots.push_back(coefficients[0] + coefficients[1] * t
-                        + coefficients[2] * std::pow(t, 2)
+        knots.push_back(coefficients[0] + coefficients[1] * t + coefficients[2] * std::pow(t, 2)
                         + coefficients[3] * std::pow(t, 3));
     }
 
     const double initT = std::chrono::duration<double>(initTime).count();
     const double finalT = std::chrono::duration<double>(finalTime).count();
 
-    Eigen::Vector4d initVelocity = coefficients[1] + 2 * coefficients[2] * initT
-                                   + 3 * coefficients[3] * std::pow(initT, 2);
+    Eigen::Vector4d initVelocity
+        = coefficients[1] + 2 * coefficients[2] * initT + 3 * coefficients[3] * std::pow(initT, 2);
 
     Eigen::Vector4d finalVelocity = coefficients[1] + 2 * coefficients[2] * finalT
                                     + 3 * coefficients[3] * std::pow(finalT, 2);
 
-    CubicSpline spline;
+    Eigen::Vector4d initAcceleration = 2 * coefficients[2] + 3 * 2 * coefficients[3] * initT;
+    Eigen::Vector4d finalAcceleration = 2 * coefficients[2] + 3 * 2 * coefficients[3] * finalT;
+
+    CubicSpline<Eigen::Vector4d> spline;
     REQUIRE(spline.setKnots(knots, time));
 
-    REQUIRE(spline.setInitialConditions(initVelocity));
-    REQUIRE(spline.setFinalConditions(finalVelocity));
+    REQUIRE(spline.setInitialConditions({initVelocity, initAcceleration}));
```

Comment:
Is this a breaking change?

Reply:
yes I will add it as deprecated

Reply:
Done in https://github.com/ami-iit/bipedal-locomotion-framework/pull/773/commits/8e37f1e43a888c7cf6fc2bed833b59d8c5d47a37